From 9ede499ffe2f9684c3037cf1eb3ea62467a32d17 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Sat, 13 Jan 2018 02:20:23 -0700
Subject: [PATCH 13/17] klte-common: HAXX: Fix seeming RIL start race condition

* In particular, the RIL on CDMA variants seems to only work reliably
  on first boot after flash, when things are all slowed down while
  dexopting. After that first boot, it's hit and miss whether a
  particular boot will have a functional radio.
* Slow things down by limiting ourselves to a single CPU on boot,
  bringing the rest online later.

Change-Id: Ie194740cf0487268dc0dbd3377bbb790cdd1b04d
---
 BoardConfigCommon.mk     | 2 +-
 rootdir/etc/init.qcom.rc | 5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index caf4da3..2f1300f 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -54,7 +54,7 @@ DEVICE_MANIFEST_FILE += $(COMMON_PATH)/manifest.xml
 
 # Kernel
 BOARD_KERNEL_BASE := 0x00000000
-BOARD_KERNEL_CMDLINE := androidboot.hardware=qcom user_debug=31 msm_rtb.filter=0x37 ehci-hcd.park=3 zcache.enabled=1 zcache.compressor=lz4
+BOARD_KERNEL_CMDLINE := androidboot.hardware=qcom user_debug=31 msm_rtb.filter=0x37 ehci-hcd.park=3 zcache.enabled=1 zcache.compressor=lz4 maxcpus=1
 BOARD_KERNEL_CMDLINE += androidboot.selinux=permissive
 BOARD_KERNEL_IMAGE_NAME := zImage
 BOARD_KERNEL_PAGESIZE := 2048
diff --git a/rootdir/etc/init.qcom.rc b/rootdir/etc/init.qcom.rc
index e8d7b5e..5fb70d7 100644
--- a/rootdir/etc/init.qcom.rc
+++ b/rootdir/etc/init.qcom.rc
@@ -255,6 +255,11 @@ on post-fs-data
     rmdir /data/data/com.android.providers.telephony/databases
     rmdir /data/data/com.android.providers.telephony/shared_prefs
 
+    write /sys/devices/system/cpu/cpu0/online 1
+    write /sys/devices/system/cpu/cpu1/online 1
+    write /sys/devices/system/cpu/cpu2/online 1
+    write /sys/devices/system/cpu/cpu3/online 1
+
     setprop vold.post_fs_data_done 1
 
 on early-boot
-- 
2.7.4

